computer_vision/haralick_descriptors.py

```diff
@@ -0,0 +1,433 @@
+            # Apply transformation method to the centered section of the image
+            transformed[x - center_x, y - center_y] = apply(center[kernel == 1])
+
+    return transformed
+
+
+def opening_filter(image: np.ndarray, kernel: np.ndarray = None) -> np.ndarray:
+    """
+    Opening filter, defined as the sequence of
+    erosion and then a dilation filter on the same image.
+
+    Examples:
+        >>> img = np.array([[1, 0.5], [0.2, 0.7]])
+        >>> img = binarize(img, threshold=0.5)
+        >>> opening_filter(img)
+        array([[1, 1],
+               [1, 1]], dtype=uint8)
+    """
+    if kernel is None:
+        np.ones((3, 3))
+
+    return transform(transform(image, "dilation", kernel), "erosion", kernel)
+
+
+def closing_filter(image: np.ndarray, kernel: np.ndarray = None) -> np.ndarray:
+    """
+    Opening filter, defined as the sequence of
+    dilation and then erosion filter on the same image.
+
+    Examples:
+        >>> img = np.array([[1, 0.5], [0.2, 0.7]])
+        >>> img = binarize(img, threshold=0.5)
+        >>> closing_filter(img)
+        array([[0, 0],
+               [0, 0]], dtype=uint8)
+    """
+    if kernel is None:
+        kernel = np.ones((3, 3))
+    return transform(transform(image, "erosion", kernel), "dilation", kernel)
+
+
+def binary_mask(
+    image_gray: np.ndarray, image_map: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Apply binary mask, or thresholding based
+    on bit mask value (mapping mask is binary).
+
+    Returns the mapped true value mask and its complementary false value mask.
+
+    Example:
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> binary_mask(gray, morphological)
+        (array([[1, 1],
+               [1, 1]], dtype=uint8), array([[158,  97],
+               [ 56, 200]], dtype=uint8))
+    """
+    true_mask, false_mask = np.array(image_gray, copy=True), np.array(
+        image_gray, copy=True
+    )
+    true_mask[image_map == 1] = 1
+    false_mask[image_map == 0] = 0
+
+    return true_mask, false_mask
+
+
+def matrix_concurrency(image: np.ndarray, coordinate: list) -> np.ndarray:
+    """
+    Calculate sample co-occurrence matrix based on input image
+    as well as selected coordinates on image.
+
+    Implementation is made using basic iteration,
+    as function to be performed (np.max) is non-linear and therefore
+    not callable on the frequency domain.
+
+    Example:
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> mask_1 = binary_mask(gray, morphological)[0]
+        >>> matrix_concurrency(mask_1, [0, 1])
```

Comment:
```suggestion
        >>> matrix_concurrency(mask_1, (0, 1))
```
Change list to tuple (see previous comments)